crates/nu-command/tests/commands/cp.rs

```diff
@@ -344,3 +347,42 @@ fn copy_ignores_ansi() {
         assert_eq!(actual.out, "success.txt");
     });
 }
+
+#[test]
+fn copy_file_not_exists_dst() {
+    Playground::setup("cp_test_17", |_dirs, sandbox| {
+        sandbox.with_files(vec![EmptyFile("valid.txt")]);
+
+        let actual = nu!(
+            cwd: sandbox.cwd(),
+            "cp valid.txt ~/invalid_dir/invalid_dir1"
+        );
+        assert!(
+            actual.err.contains("invalid_dir1")
+                && (actual.err.contains("No such file or directory")
+                    || actual
+                        .err
+                        .contains("The system cannot find the path specified"))
+        );
+    });
+}
+
+#[test]
+fn copy_file_with_read_permission() {
+    Playground::setup("cp_test_18", |_dirs, sandbox| {
+        sandbox.with_files(vec![
+            EmptyFile("valid.txt"),
+            FileWithPermission("invalid_prem.txt", false),
+        ]);
+
+        let actual = nu!(
+            cwd: sandbox.cwd(),
+            "cp valid.txt invalid_prem.txt",
+        );
+        assert!(
+            actual.err.contains("invalid_prem.txt")
+                && (actual.err.contains("Access is denied")
+                    || actual.err.contains("Permission denied"))
```

Comment:
Again, make sure that we try to avoid having tests that might fail based on a developers OS.